docs/getting-started-guides/kubeadm.md

```diff
@@ -54,358 +65,556 @@ If you are not constrained, there are some other tools built to give you complet
 
-You can now access the API Server locally at `http://localhost:8001/api/v1`
+deb-packages are released for ARM and ARM 64-bit, but not RPMs (yet, reach out
+if there's interest).
 
-### (Optional) Installing a sample application
+Currently, only the pod network flannel is working on multiple architectures.
+You can install it this way:
 
-As an example, install a sample microservices application, a socks shop, to put your cluster through its paces. Note that this demo does only work on `amd64`.
-To learn more about the sample microservices app, see the [GitHub README](https://github.com/microservices-demo/microservices-demo).
+``` bash
+export ARCH=amd64
+curl -sSL "https://github.com/coreos/flannel/blob/master/Documentation/kube-flannel.yml?raw=true" | sed "s/amd64/${ARCH}/g" | kubectl create -f -
+```
 
-    # kubectl create namespace sock-shop
-    # kubectl apply -n sock-shop -f "https://github.com/microservices-demo/microservices-demo/blob/master/deploy/kubernetes/complete-demo.yaml?raw=true"
+Replace `ARCH=amd64` with `ARCH=arm` or `ARCH=arm64` depending on the platform
+you're running on. Note that the Raspberry Pi 3 is in ARM 32-bit mode, so for
+RPi 3 you should set `ARCH` to `arm`, not `arm64`.
 
-You can then find out the port that the [NodePort feature of services](/docs/user-guide/services/) allocated for the front-end service by running:
+## Cloudprovider integrations (experimental)
 
-    # kubectl describe svc front-end -n sock-shop
-    Name:                   front-end
-    Namespace:              sock-shop
-    Labels:                 name=front-end
-    Selector:               name=front-end
-    Type:                   NodePort
-    IP:                     100.66.88.176
-    Port:                   <unset> 80/TCP
-    NodePort:               <unset> 31869/TCP
-    Endpoints:              <none>
-    Session Affinity:       None
+Enabling specific cloud providers is a common request. This currently requires
+manual configuration and is therefore not yet fully supported. If you wish to do
+so, edit the kubeadm dropin for the kubelet service
+(`/etc/systemd/system/kubelet.service.d/10-kubeadm.conf`) on all nodes,
+including the master. If your cloud provider requires any extra packages
+installed on host, for example for volume mounting/unmounting, install those
+packages.
+
+Specify the `--cloud-provider` flag to kubelet and set it to the cloud of your
+choice. If your cloudprovider requires a configuration file, create the file
+`/etc/kubernetes/cloud-config` on every node. The exact format and content of
+that file depends on the requirements imposed by your cloud provider. If you use
+the `/etc/kubernetes/cloud-config` file, you must append it to the kubelet
+arguments as follows: `--cloud-config=/etc/kubernetes/cloud-config`
+
+Next, specify the cloud provider in the kubeadm config file.  Create a file called
+`kubeadm.conf` with the following contents:
+
+``` yaml
+cloudProvider: <cloud provider>
```

Comment:
please make this a full example:
```yaml
kind: MasterConfiguration
apiVersion: kubeadm.k8s.io/v1alpha1
cloudProvider: <cloud provider>
```

and point out that **this** part (the config file) of kubeadm is alpha